forem/forem
# Refactor editor tags selector to fetch all tag info in one network call

## What type of PR is this? (check all applicable)


## Description

We previously introduced this change in https://github.com/forem/forem/pull/16610, but we had to revert it due to issues with caching on the new endpoint (which caused top tags to be returned instead of the ones which were searched for).

This PR builds on work in https://github.com/forem/forem/pull/16671 which amends the bulk tags endpoint so that we can safely make this refactor.

## Related Tickets & Documents



Related Issues:
- https://github.com/forem/forem/pull/16671
- https://github.com/forem/forem/pull/16610

Closes https://github.com/forem/forem/issues/16540

## QA Instructions, Screenshots, Recordings

The current Cypress specs test this functionality, but to test manually:

- Create a post with multiple supported tags (and try an unsupported one too)
- Save the post
- Click to edit the post and check that the correct tags appear in the editor, along with the correct colours etc
- Check that only one network call was made to fetch those tags

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] No, and this is why: This is a straight refactor, and the existing tests continue to verify the behaviour

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- The related issue is tagged for a changelog rollup

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Lets try again](https://media.giphy.com/media/kKLmEOQy45G4QzFPOO/giphy.gif)
